op-node/rollup/derive/engine_queue.go

```diff
@@ -683,6 +706,51 @@ func (eq *EngineQueue) forceNextSafeAttributes(ctx context.Context) error {
 	return nil
 }
 
+// tryBackupUnsafeReorg tries to reorg unsafe head to backupUnsafeHead.
+func (eq *EngineQueue) tryBackupUnsafeReorg(ctx context.Context) error {
+	defer func() {
+		// Reset backupUnsafeHead to avoid multiple usage.
+		eq.backupUnsafeHead = eth.L2BlockRef{}
+		// Only try once because execution engine may forgot backupUnsafeHead
+		// or backupUnsafeHead is not part of the chain.
+		eq.checkBackupUnsafeReorg = false
+	}()
+	if eq.backupUnsafeHead == (eth.L2BlockRef{}) { // sanity check backupUnsafeHead is there
+		return nil
+	}
+	if eq.backupUnsafeHead.Number <= eq.unsafeHead.Number {
+		eq.log.Info("current safe head is already the longest unsafe chain", "backupUnsafe", eq.backupUnsafeHead.ID(), "unsafe", eq.unsafeHead.ID())
+		return nil
+	}
+	// Reorg unsafe chain. Safe/Finalized chain will not be updated.
+	eq.log.Warn("trying to restore unsafe head", "backupUnsafe", eq.backupUnsafeHead.ID(), "unsafe", eq.unsafeHead.ID())
+	fc := eth.ForkchoiceState{
+		HeadBlockHash:      eq.backupUnsafeHead.Hash,
```

Comment:
As I understand, this method must not be called when the rollup node is doing EL sync. So we don't have to consider engineSyncTarget here. But just in case, why don't you check if the node is in progress of EL sync (engineSyncTarget != unsafeHead)?